### PR TITLE
Mock Backend for unittests

### PR DESCRIPTION
**Pull request by kratos7**
Mock backened that mimics marathon API for hdyra infra to consume.

Same test can now be run on mesos/marathon as well as local host by setting mock=True
for RunTestRMQ.

Added  a unittest   rmq_local_test.py:
To run:
python rmq_local_test.py

Note:
Only added a unit test for RMQTest for now.
Also, for now, all the child processes logs go into hydra/live directory.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -29,5 +28,11 @@ install:
   - pyb install_dependencies analyze --verbose
 
 script:
-  - pyb install
+  - pyb publish -x run_unit_tests
+  - rm src/main/scripts/zmq_*
```
Comment (sushilks) [inline .travis.yml]: This is not needed as you got rid for 3.5 support.

File: .travis.yml [hunk 2]
```diff
@@ -29,5 +28,11 @@ install:
...
+  - pyb install -x run_unit_tests
   - pyb test --verbose
+
+after_script:
+  - cat /home/travis/build/sushilks/hydra/live/*.log
+  - cat /home/travis/build/sushilks/hydra/target/reports/distutils/bdist_wheel
```
Comment (sushilks) [inline .travis.yml]: also this line can be removed.

File: src/main/python/hydra/lib/mock_backend.py [hunk 3]
```diff
@@ -0,0 +1,254 @@
...
+        to be passed on to child processes as ports
+
+        TODO (AbdullahS): There is no guarantee that
+                          generated ports will be available
+                          at OS level as well. See if this can
+                          be improved.
+        """
+        self.env_ports = []
+        for x in range(self.total_ports):
+            self.env_ports.append(random.randrange(10000, 20000))
```
Comment (sushilks) [inline src/main/python/hydra/lib/mock_backend.py:82]: OS will not use ports between 2000 to 23767 so you should be safe

File: src/main/python/hydra/rmqtest/rmq_sub.py [hunk 4]
```diff
@@ -58,7 +58,11 @@ def run10(argv):
     pwd = os.getcwd()
     l.info("CWD = " + pformat(pwd))
     cmgr = ChildManager()
+    myenv = os.environ.copy()
     cmd = './hydra hydra.rmqtest.rmq_sub.run'.split(' ') + argv[1:]
+    # TODO: (AbdullahS) Find a better way to do this
+    if "mock" in myenv:
```
Comment (sushilks) [inline src/main/python/hydra/rmqtest/rmq_sub.py:64]: why is this needed ?

File: src/main/python/hydra/rmqtest/runtest.py [hunk 5]
```diff
@@ -16,7 +16,7 @@
     from configparser import ConfigParser
 
 l = util.createlogger('runTest', logging.INFO)
-# l.setLevel(logging.DEBUG)
+l.setLevel(logging.DEBUG)
```
Comment (sushilks) [inline src/main/python/hydra/rmqtest/runtest.py]: probably not needed

File: src/main/python/hydra/rmqtest/rmq_sub.py (hunk 4 above)
Comment (kratos7) [inline src/main/python/hydra/rmqtest/rmq_sub.py:64]: ./ is run from the target/src/.. directory on remote slaves in case of mesos/marathon, the same path is not true for local test so the command has to be modified, hydra is in the path, so just hydra works.

**Comment by sushilks** [comment]
Woo Hoo!!! We have our first real test case ... :+1:
